ml-algorithms/src/test/java/org/opensearch/ml/engine/MLEngineTest.java

```diff
@@ -55,11 +56,14 @@ public class MLEngineTest {
     public ExpectedException exceptionRule = ExpectedException.none();
 
     private MLEngine mlEngine;
+    private MockSettings mockSettingsWithInlineMockMaker;
 
     @Before
     public void setUp() {
         Encryptor encryptor = new EncryptorImpl("m+dWmfmnNRiNlOdej/QelEkvMTyH//frS2TBeS2BP4w=");
         mlEngine = new MLEngine(Path.of("/tmp/test" + UUID.randomUUID()), encryptor);
+        mockSettingsWithInlineMockMaker = new MockSettingsImpl()
```

Comment:
Is it possible to put this in a configuration file and all test files can use mockStatic method without the boilerplate code?

Reply:
OpenSearch-core overrides the default mock maker through mockito-extension which limits the use of mockStatic. We can provide our own mockito-extension, but it'll impact both mock and mockStatic.

Reply:
What impact will it be for using our own mockito-extension?

Reply:
> if several mockito-extensions/org.mockito.plugins.MockMaker files exists in the classpath Mockito will only use the first returned by the standard [ClassLoader.getResource(java.lang.String)](http://docs.oracle.com/javase/8/docs/api/java/lang/ClassLoader.html?is-external=true#getResource(java.lang.String)) mechanism.

https://site.mockito.org/javadoc/current/org/mockito/plugins/MockMaker.html

So technically, if a library depends on both ml-commons and opensearch-core (might not be a case for now), could possibly start to use ml-commons's mockito-extension which changes their behavior implicitly. I still feel ideally, the fix should be done in opensearch-core to support both normal mock maker and inline mock maker, but since maintainers don't want to do in core, and mockStatic is still rare case, I feel we can just make mockStatic work in this way.

Reply:
Ok, is it true that creating an abstract class like `StaticMockSupportTest` and a method like `myMockStatic`, then all the sub-classes of it can use myMockStatic method directly?

Reply:
Made changes